mycroft/skills/core.py

```diff
@@ -699,9 +712,44 @@ def cancel_event(self, name):
             Args:
                 name (str):   Name of event
         """
-        data = {'event': self._unique_name(name)}
+        unique_name = self._unique_name(name)
+        data = {'event': unique_name}
+        self.remove_event(unique_name)
         self.emitter.emit(Message('mycroft.scheduler.remove_event', data=data))
 
+    def get_event_status(self, name):
+        """
+            Get event data and return the amount of time left
+
+            Args:
+                name (str): Name of event
+
+            Return:
+                (datetime, int) (tuple): datetime and time left in epoch time
+        """
+        event_name = self._unique_name(name)
+        data = {'name': event_name}
+
+        # this is needed because message.data could return None
+        # and that would make the while loop infinite
+        self.event_status = "None"
+
+        def callback(message):
+            event_time = int(message.data[0][0])
+            current_time = int(time.time())
+            time_left_in_seconds = event_time - current_time
+            date = datetime.datetime.now() + time_left
+            self.event_status = (date, time_left_in_seconds)
+
+        emitter_name = 'mycroft.event_status.callback.{}'.format(event_name)
+
+        self.emitter.once(emitter_name, callback)
+        self.emitter.emit(Message('mycroft.scheduler.get_event', data=data))
+
+        while True:
```

Comment:
This should probably have a timeout of say two seconds, and do sleeps between checks to avoid hogging the cpu:

```python
start_wait = time.time()
while event_status is None and time.time() - start_wait < 2.0:
    time.sleep(0.1)
return event_status
```
(Having it return `None` on failure is a common approach in python)

Reply:
Even better would be if it raised an exception on timeout. Not sure which one would be approprate. IOError or some from the websocket client?